File: src/main/java/org/kd/main/common/entities/Transfer.java
package org.kd.main.common.entities;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import net.bytebuddy.implementation.bytecode.constant.LongConstant;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Objects;

@Entity
@Getter
@Setter
@EqualsAndHashCode
public class Transfer implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private Long src_customer_id;
    private Long dest_customer_id;
    private Float units;
    private Boolean internal;

    public Transfer(long src_customer_id, long dest_customer_id, float units, boolean internal) {
        this.src_customer_id = src_customer_id;
        this.dest_customer_id = dest_customer_id;
        this.units = units;
        this.internal = internal;
    }
}

File: C#UI/Banque/Common/Entities/Transfer.cs
namespace Banque.Common.Entities
{
    public class Transfer
    {
        public long Id { get; set; }
        public Account srcAccount { get; set; }
        public Account destAccount { get; set; }
        public double units { get; set; }
        public bool internalTransfer  { get; set; }

        public Transfer(Account srcAccount, Account destAccount, double units, bool internalTransfer) {
        this.srcAccount = srcAccount;
        this.destAccount = destAccount;
        this.units = units;
        this.internalTransfer = internalTransfer;
    }
}
}
